microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks.md

```diff
@@ -0,0 +1,120 @@
+
+The **Sources** tool opens in the top panel to line 13 of the file **room.js**.
+
+1. In the `hide()` function of **room.js**, you can see that the JavaScript code of the demo application adds each message in the room to an `unmounted` array. The `unmounted` array is the object that is referencing the detached element.
+
+:::image type="complex" source="./images/detached-elements-javascript-retainers.msft.png" alt-text="Identifying the JavaScript that is retaining the detached element." lightbox="./images/detached-elements-javascript-retainers.msft.png":::
+   Identifying the JavaScript that is retaining the detached element.
+:::image-end:::
+
+You have now identified the retainer that is preventing the detached element from being garbage collected by the browser!
+
+## Next steps
+
+When looking for memory leaks, remember that leaks can depend on the context of your application. For the demo application, you found detached elements that could not be garbage collected by the browser and identified the JavaScript retaining the detached elements. However, in the context of the demo application, it makes sense to retain the list of chat messages so that if a user switches back to **Room 1**, the message log is preserved.
+
+:::image type="complex" source="./images/detached-elements-reattached.msft.png" alt-text="The detached elements are reattached to the DOM when switching back to Room 1." lightbox="./images/detached-elements-reattached.msft.png":::
+   The detached elements are reattached to the DOM when switching back to Room 1.
+:::image-end:::
+
+Similarly, a feed in social media might detach elements as users scroll past them and reattach them to the DOM when users scroll back up. Detached elements are not always an indication of a memory leak and memory leaks are not always caused by detached elements.
+
+For long-running apps, small memory leaks of only a few kilobytes can noticeably degrade performance over time. Web developers who use the React framework know that React maintains a virtualized copy of the DOM. Failing to properly unmount components can potentially lead to an application leaking large parts of the virtual DOM.
+
+The demo app and leaks were artificial. Test the tool on your production website or app to see if you can find potential problems. If you do, [contact the Microsoft Edge DevTools team](../contact.md) to send feedback about the **Detached Elements** tool and memory leak debugging.
+
+## Additional features
+
+* In the **Detached Elements** tool, select the **Detach Elements** button to destroy the parent-child links inside a detached tree. Select **Collect garbage** to identify the culprit node in a detached tree that is causing the entire tree to be retained.
+
+* In the **Detached Elements** tool, change the target in the **Selected target** dropdown to select a different origin. Use this dropdown to select different frames or origins for the tool.
```

Comment:
These 2 features could be in their own h2 sections. 

```
## Identify the DOM node causing others to be retained

Because the DOM is a fully connected graph, when one DOM node is retained in memory by JavaScript, it can cause other DOM nodes to be retained with it.

To identify the culprit node is a detached tree that is causing the entire tree to be retained:

* Click **Detach elements** to destroy the parent-child links inside the detached tree.
* Click **Collect garbage**.
```

And then you can perhaps add a screenshot with red boxes around the buttons that need to be clicked.

Same thing for the target selection feature.